lib/HLSL/DxilValidation.cpp

```diff
@@ -5850,8 +5850,8 @@ HRESULT ValidateDxilBitcode(
     IFT(CreateMemoryStream(DxcGetThreadMallocNoRef(), &pOutputStream));
     pOutputStream->Reserve(pWriter->size());
     pWriter->write(pOutputStream);
+    const DxilVersionedRootSignatureDesc* pDesc = nullptr;
```

Comment:
Could we create a small object wrapper for this which owns (and automatically frees) the memory, then has a `const DxilVersionedRootSignatureDesc*` cast operator, a `*` operator, and `get()` method (all returning this type)?  That would make it more reliable to work with and avoid memory leaks, right?

Reply:
And, perhaps a non-const `get()` or `get_mutable()` could return the non-const pointer, allowing you to modify it without using const_cast (as long as you promise to be careful with memory management 😉).

Reply:
Is this stuff visible to the driver? Is that why it's binary-stable? Or is it just because it's trying to echo the API/DDI structures? If it's the latter, is there any reason not to replace the bare arrays with std::vector?

Reply:
Not so sure about the suggestion of allowing the callers to get non-const access to the bare arrays, but it's your guys' call. It would allow me to isolate the code that modifies the arrays inside the PIX passes.

Reply:
Given that Tex+Greg advocate doing the memory allocation outside of this file (and the attendant de-duplication of some of this new code), I've done so. And added the wrapper Tex suggested.